openbrews/recipe/RecipeUtils.js

```diff
@@ -0,0 +1,131 @@
+angular.module('openbrews.recipeUtils', [])
+.service('RecipeUtils', function () {
+
+    // calculates IBU given a recipe and OG (original gravity).
+    this.calculateIbu = function (recipe, og) {
+        var hops = recipe.hops.filter(function (hop) {
+            return hop.aa;
+        });
+
+        var gravity = this.boilGravity(recipe, og);
+
+        return hops.reduce(function (acc, hop) {
+            return acc + this.ibuFromHop(hop, recipe.batchSize, gravity) / hops.length;
+        }, 0);
+    }
+
+    this.ibuFromHop = function (hop, batchSize, gravity) {
```

Comment:
Sure, I'll see what I can do.

According to [this](http://howtobrew.com/book/section-1/hops/hop-bittering-calculations) page, it seems like the numbers were just 'found' by recording a lot of data and don't really represent anything. I'm not sure we should make them constants or not?

I've included a link to the website as a comment in the code.